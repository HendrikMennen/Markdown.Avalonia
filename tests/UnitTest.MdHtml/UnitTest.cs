using ApprovalTests;
using ApprovalTests.Reporters;
using Avalonia.Controls;
using Markdown.Avalonia;
using Markdown.Avalonia.Html;
using NUnit.Framework;
using System.Linq;
using UnitTest.Base;
using UnitTest.Base.Utils;
using UnitTest.MdHtml.Test;

using MarkdownEngine = Markdown.Avalonia.Markdown;

namespace UnitTest.MdHtml
{
    [UseReporter(typeof(DiffReporter))]
    public class UnitTest : UnitTestBase
    {
        public MarkdownEngine Engine
        {
            get
            {
                var plugins = new MdAvPlugins();
                plugins.Plugins.Add(new HtmlPlugin());

                return new MarkdownEngine()
                {
                    Plugins = plugins
                };
            }
        }

        [Test]
        [RunOnUI]
        public void Button()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void CodeBlock()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void InlineCode()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void Input()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void List()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void Progres()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void Details()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void TypicalBlock()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void TypicalInline()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void Mixing()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }
    }
}
