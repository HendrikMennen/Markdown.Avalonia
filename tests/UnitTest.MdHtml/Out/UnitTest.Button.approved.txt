<?xml version="1.0" encoding="utf-16"?>
<StackPanel xmlns:c="clr-namespace:ColorTextBlock.Avalonia;assembly=ColorTextBlock.Avalonia" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" xmlns="https://github.com/avaloniaui">
  <c:CTextBlock>
    <c:CInlineUIContainer>
      <c:CInlineUIContainer.Content>
        <Button IsEnabled="False" Classes=":disabled" xmlns="">
          <Button.Content>
            <StackPanel />
          </Button.Content>
        </Button>
      </c:CInlineUIContainer.Content>
    </c:CInlineUIContainer>
    <c:CRun Text=" " />
    <c:CInlineUIContainer>
      <c:CInlineUIContainer.Content>
        <Button IsEnabled="False" Classes=":disabled" xmlns="">
          <Button.Content>
            <StackPanel />
          </Button.Content>
        </Button>
      </c:CInlineUIContainer.Content>
    </c:CInlineUIContainer>
    <c:CRun Text=" " />
    <c:CInlineUIContainer>
      <c:CInlineUIContainer.Content>
        <Button IsEnabled="False" Classes=":disabled" xmlns="">
          <Button.Content>
            <StackPanel>
              <c:CTextBlock>
                <c:CRun Text="some text" />
              </c:CTextBlock>
            </StackPanel>
          </Button.Content>
        </Button>
      </c:CInlineUIContainer.Content>
    </c:CInlineUIContainer>
    <c:CRun Text=" " />
    <c:CInlineUIContainer>
      <c:CInlineUIContainer.Content>
        <Button IsEnabled="False" Classes=":disabled" xmlns="">
          <Button.Content>
            <StackPanel>
              <Border Classes="Center">
                <Border.Child>
                  <StackPanel Classes="Center">
                    <c:CTextBlock Classes="Heading1">
                      <c:CRun Text="title" />
                    </c:CTextBlock>
                  </StackPanel>
                </Border.Child>
              </Border>
              <c:CTextBlock>
                <c:CRun Text="description1 description2" />
              </c:CTextBlock>
            </StackPanel>
          </Button.Content>
        </Button>
      </c:CInlineUIContainer.Content>
    </c:CInlineUIContainer>
  </c:CTextBlock>
</StackPanel>
